tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/member_access.bal

```diff
@@ -771,6 +771,41 @@ public function testNestedMemberAccessOnIntersectionTypes() {
     assertEquality((), v3);
 }
 
+type myrec record {|
+    string name;
+|};
+
+const LENGTH = 10;
+
+function testMemberAccessWithBinaryExprAsIndex() {
+    int[3] intArr = [0, 1, 2];
+    assertEquality(1, intArr[10 - 9]);
+    
+    int x = 8;
+    assertEquality(2, intArr[10 - x]);
+    
+    assertEquality(0, intArr[10 - LENGTH]);
+    
+    myrec myRecVar = {name: "Ballerina"};
+    assertEquality("Ballerina", myRecVar["na" + "me"]);
```

Comment:
When we fix https://github.com/ballerina-platform/ballerina-lang/issues/33090 and https://github.com/ballerina-platform/ballerina-lang/issues/34188 the static type of some of these will change. So for other cases also shall we add tests when at least one of the operands in the binary expression is not of a singleton type? Like L785.